controller/common/lut.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Union
+
+import numpy as np
+from scipy import interpolate
+
+
+class LUT:
+    """
+    Class for performing look-up table interpolation.
+
+    Methods:
+        __init__: Initializes the LUT object with lookup table data and interpolation method.
+        __call__: Calls the interpolation method with the given input.
+    """
+
+    def __init__(
+        self,
+        lookup_table: Union[List[List[int | float]], np.ndarray],
+        interpolation_method: str = "linear",
+    ):
+        """
+        Initializes the LUT object.
+
+        Args:
+            lookup_table (list): A list of tuples or lists containing x-y data points for
```

Comment:
Your typing doesn't comply with this docstring. You say list or tuple, but there is a difference between lists and tuples. I recommend to just say list of lists. Also in the parentheses, it should match what you have in the function signature. In this case, it should be `(List[List[Scalar]] | NDArray)`.

Also, mention that the shape of the input should be `(n, 2)`

Reply:
updated docstring:

```
"""
        Initializes the LUT object.

        Args:
            lookup_table (List[List[Scalar]]|NDArray): A list of lists or NDArray containing x-y
            data points for interpolation. Shape should be (n, 2)
```